### ISSUE TITLE
SNGP | Laplace RF Precision update inconsistent with the likelihood 

### ISSUE THREAD

**Issue by nikhil-dce** [google/uncertainty-baselines#258]
For the [CIFAR10 example](https://github.com/google/uncertainty-baselines/blob/master/baselines/cifar/sngp.py), the precision matrix update (in the [LaplaceRandomFeatureCovariance](https://github.com/google/edward2/blob/master/edward2/tensorflow/layers/random_feature.py#L248) class in Edward) is based on gaussian likelihood, whereas in the loss function cross-entropy is being used. 

The batch precision update seems different than in the [SNGP paper](https://arxiv.org/pdf/2006.10108.pdf). The update should be based on equation 9 of the paper. Is this observation correct or am I missing something? 

#question

**Comment by dustinvtran** [comment]
hey @nikhil-dce! were you able to get this resolved? ccing @jereliu

**Comment by jereliu** [comment]
Hi @nikhil-dce ! 

Yes you are correct the released implementation in UB used Gaussian likelihood - the main reason is to keep code simple, and performance-wise we did not observe a significant difference in ECE (and it seems to hurt OOD a bit).

**Comment by nikhil-dce** [comment]
Hi @jereliu - Thank you for getting back to me and answering my question. It might be nice to add a comment somewhere in the paper/implementation regarding this deviation from the paper. Thanks again!

> Hi @nikhil-dce !

> Yes you are correct the released implementation in UB used Gaussian likelihood - the main reason is to keep code simple, and performance-wise we did not observe a significant difference in ECE (and it seems to hurt OOD a bit).

**Comment by jereliu** [comment]
Thanks Nikhil! Yes will add a note to the implementation in an upcoming code update. Will report here when the pull request merged.

> Hi @jereliu - Thank you for getting back to me and answering my question. It might be nice to add a comment somewhere in the paper/implementation regarding this deviation from the paper. Thanks again!
> 
> > Hi @nikhil-dce !
> 
> > Yes you are correct the released implementation in UB used Gaussian likelihood - the main reason is to keep code simple, and performance-wise we did not observe a significant difference in ECE (and it seems to hurt OOD a bit).
